macaroons/service_test.go

```diff
@@ -61,17 +61,17 @@ func TestNewService(t *testing.T) {
 	// Second, create the new service instance, unlock it and pass in a
 	// checker that we expect it to add to the bakery.
 	service, err := macaroons.NewService(tempDir, macaroons.IPLockChecker)
-	defer service.Close()
 	if err != nil {
 		t.Fatalf("Error creating new service: %v", err)
 	}
+	defer service.Close()
 	err = service.CreateUnlock(&defaultPw)
 	if err != nil {
 		t.Fatalf("Error unlocking root key storage: %v", err)
 	}
 
 	// Third, check if the created service can bake macaroons.
-	macaroon, err := service.Oven.NewMacaroon(nil, bakery.LatestVersion,
+	macaroon, err := service.Oven.NewMacaroon(context.TODO(), bakery.LatestVersion,
```

Comment:
nit: line gets too long with the change.

Reply:
@guggero Done.

Reply:
@alrs thanks!
We should describe this in the contribution guidelines, but we have a specific way of formatting argument lists of long method calls.
The code you changed is pretty old and therefore wasn't yet properly formatted to the new agreed upon style. So it would be nice if you could fix just these two calls to `NewMacaroon` to the new style. Sorry I missed this before.

What I mean is:
```
	macaroon, err := service.Oven.NewMacaroon(
		context.TODO(), bakery.LatestVersion, nil, testOperation,
	)
```

Reply:
Done.